### PR TITLE
fix(tooltip): improve component timing

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #6396

## Summary

- Separates timeout delay between open and close.
  - `open` is set to `300ms`.
  - `close` is set to `500ms`.
- Sets delay to `0ms` if a tooltip is already displayed.
- Add story for design review.
- Update tests.

### CODE REVIEW

**Comment by driskull** [comment]
@ashetland can you review these changes? 

https://6266d45509d7eb004aa254fb-pnvhhobnly.chromatic.com/?path=/story/components-action-bar--with-slotted-tooltips

**Comment by ashetland** [comment]
> @ashetland can you review these changes?
> 
> https://6266d45509d7eb004aa254fb-pnvhhobnly.chromatic.com/?path=/story/components-action-bar--with-slotted-tooltips

This is great @driskull! Thanks for your work on this!

**Comment by driskull** [comment]
@ashetland what about the animation on the tooltips. Do we need to do any work to disable them when "instant"?

File: packages/calcite-components/src/components/tooltip/TooltipManager.ts [hunk 1]
```diff
@@ -192,25 +204,37 @@ export default class TooltipManager {
   private toggleTooltip(tooltip: HTMLCalciteTooltipElement, value: boolean): void {
```
Comment (jcfranco) [inline packages/calcite-components/src/components/tooltip/TooltipManager.ts:204]: Can you rename `value` to convey its purpose?

File: packages/calcite-components/src/components/tooltip/TooltipManager.ts [hunk 2]
```diff
@@ -192,25 +204,37 @@ export default class TooltipManager {
...
+        this.closeActiveTooltip();
 
-      this.closeExistingTooltip();
+        if (tooltip !== this.hoveredTooltip) {
+          return;
+        }
+
+        this.toggleTooltip(tooltip, true);
+      },
+      this.activeTooltip ? 0 : TOOLTIP_OPEN_DELAY_MS
```
Comment (jcfranco) [inline packages/calcite-components/src/components/tooltip/TooltipManager.ts:226]: Could you add a test for the `0 delay when there's an active tooltip`?

File: packages/calcite-components/src/components/action-bar/action-bar.stories.ts [hunk 3]
```diff
@@ -148,6 +148,36 @@ export const darkModeRTL_TestOnly = (): string =>
 
 darkModeRTL_TestOnly.parameters = { modes: modesDarkDefault };
 
+export const withSlottedTooltips = (): string => html`<div style="display:flex; height:500px; width: 200px;">
```
Comment (jcfranco) [inline packages/calcite-components/src/components/action-bar/action-bar.stories.ts:151]: This story seems unrelated to the timing changes. Can you submit them in a separate `test` type PR?

**Review by jcfranco — APPROVED** [review]
Awesome! 💬

File: packages/calcite-components/src/components/action-bar/action-bar.stories.ts (hunk 3 above)
Comment (driskull) [reply packages/calcite-components/src/components/action-bar/action-bar.stories.ts:151]: Its related to the timing changes to see instant tooltips once one is already open. Its the main request use case for instant tooltips.

Comment (jcfranco) [reply packages/calcite-components/src/components/action-bar/action-bar.stories.ts:151]: I see. Could you update the story name to reflect the main requested use case then?
